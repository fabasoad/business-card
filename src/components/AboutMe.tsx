/// <reference path='./AboutMe.d.ts' />
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import * as moment from 'moment';

import imgAboutMe from "../assets/images/about_me.jpg";

export default function AboutMe(props) {
  const { t } = useTranslation();

  const now : moment.Moment = moment();

  const startWorkDate : moment.Moment = moment('2010-03-01', 'YYYY-MM-DD');
  const duration1 : moment.Duration = moment.duration(now.diff(startWorkDate));

  const agileWorkDate : moment.Moment = moment('2011-10-01', 'YYYY-MM-DD');
  const duration2 : moment.Duration = moment.duration(now.diff(agileWorkDate));

  const generalListItem1Keys = {
    "totalExperience": duration1.locale(props.getLocale().code).humanize(),
    "agileExperience": duration2.locale(props.getLocale().code).humanize()
  };

  return (
    <div id="about">
      <div className="container">
        <div className="section-title text-center center">
          <h2>{t('business-card-about-me-title')}</h2>
          <hr />
        </div>
        <div className="row d-flex justify-content-center">
          <div className="col-md-12 text-center">
            <img className="img-thumbnail" src={imgAboutMe} alt="About me" />
          </div>
          <div className="col-md-8 col-md-offset-2 text-center">
            <div className="about-text">
              <p>{t("business-card-about-me-general-title")}</p>
              <ul className="list-unstyled">
                <li>{t("business-card-about-me-general-list-item-1", generalListItem1Keys)}</li>
                <li>{t("business-card-about-me-general-list-item-2")}</li>
                <li>{t("business-card-about-me-general-list-item-3")}</li>
                <li>{t("business-card-about-me-general-list-item-4")}</li>
                <li>{t("business-card-about-me-general-list-item-5")}</li>
                <li>{t("business-card-about-me-general-list-item-6")}</li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}