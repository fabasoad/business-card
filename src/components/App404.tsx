import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'
import { Locale, AppActions } from '../store/locale/types'
import { AppState } from '../store/configureStore'
import BaseConstants from '../scripts/BaseConstants'

interface App404Props {
  redirectUrl: string
}

type Props = App404Props & LinkDispatchProps & LinkStateProps

export function App404(props: Props) {
  const { t } = useTranslation()

  // eslint-disable-next-line no-undef
  let timer: NodeJS.Timeout
  let counterValue: number = 3

  const [isRedirecting, setIsRedirecting] = React.useState<boolean>(false)
  const [counter, setCounter] = React.useState<number>(counterValue)

  document.addEventListener('click', (e: MouseEvent) => {
    e.preventDefault()
    setIsRedirecting(true)
    timer = setInterval(() => {
      if (counterValue == 1) {
        clearInterval(timer)
        window.location.replace(props.redirectUrl)
      } else {
        counterValue--
        setCounter(counterValue)
      }
    }, 1000)
  })

  const isRedirectingStyle = (flag: boolean) => ({ display: `${isRedirecting === flag ? 'inline' : 'none'}` })

  return (
    <div className={`font-regular font-${props.locale.code === 'jp' ? '' : 'non-'}jp`}>
      <p>{t('business-card-404-text-1')}</p>
      <p>
        <span className="blinker-prefix"></span>
        {/* JP: 画面をクリックして続ける */}
        &nbsp;{t('business-card-404-text-2')}<span style={isRedirectingStyle(true)}>.</span>
        <span style={isRedirectingStyle(false)} className="blinker">_</span>
      </p>
      <p style={isRedirectingStyle(true)}>{t('business-card-404-text-3')}<span>{counter}</span><span className="blinker">_</span></p>
    </div>
  )
}

App404.defaultProps = {
  redirectUrl: BaseConstants.BASE_URL
}

interface LinkStateProps {
  locale: Locale
}

interface LinkDispatchProps {
}

const mapStateToProps = (
  state: AppState,
  ownProps: App404Props
): LinkStateProps => ({
  locale: state.locale
})

const mapDispatchToProps = (
  dispatch: ThunkDispatch<any, any, AppActions>,
  ownProps: App404Props
): LinkDispatchProps => ({
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App404)
