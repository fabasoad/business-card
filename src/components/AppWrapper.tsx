import * as React from 'react'
import Error from './Controls/Error'
import LoadingSpinner from './LoadingSpinner'
import i18nService, { I18nServiceCallbackTypes } from '../scripts/i18n/I18nService'
import producer from '../scripts/DevMessageProducer'

enum Status {
  LOADING,
  FINISHED,
  ERROR
}

export default function AppWrapper({ children }: React.PropsWithChildren<{}>) {
  const [status, setStatus] = React.useState<Status>(Status.LOADING)
  i18nService.registerCallback(
    I18nServiceCallbackTypes.ON_LOADED,
    () => setStatus(Status.FINISHED)
  )

  producer.run()

  React.useEffect(() => {
    i18nService.load().catch((e: any) => {
      setStatus(Status.ERROR)
      console.error(e)
    })
  }, [])

  switch (status) {
    case Status.LOADING:
      return <LoadingSpinner />
    case Status.FINISHED:
      return <>{children}</>
    default:
      return (
        <Error title="Failed to load i18n settings">
          <a target="_blank" rel="noopener noreferrer" href="https://github.com/fabasoad/business-card/issues/new/choose">Please raise GitHub issue</a>
        </Error>
      )
  }
}
