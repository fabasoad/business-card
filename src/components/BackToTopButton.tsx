import * as React from 'react'

export default function BackToTopButton() {
  const [isVisible, setIsVisible] = React.useState<boolean>(false)

  window.addEventListener(
    'scroll',
    () => setIsVisible(window.scrollY > 300),
    { passive: true }
  )

  const handleClick = (e) => {
    e.preventDefault()
    window.scrollTo({
      top: 0,
      left: 0,
      behavior: 'smooth'
    })
  }

  return (
    <a onClick={handleClick} className={(isVisible ? 'show ' : '') + 'back-to-top'}></a>
  )
}
