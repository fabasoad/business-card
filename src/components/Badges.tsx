import * as React from 'react'
import * as imgDigitalOceanBadge from '../assets/images/badges/digital-ocean-forest-2021.png'
import SectionTitle from './Controls/SectionTitle'
import { Col, Container, Row } from 'react-bootstrap'
import { useTranslation } from 'react-i18next'

export default function Badges() {
  const { t } = useTranslation()
  return (
    <div id="badges" className="light-component">
      <SectionTitle>{t('business-card-badges-title')}</SectionTitle>
      <Container>
        <Row>
          <Col sm md lg className="text-center">
            <a href="https://tree-nation.com/profile/digitalocean" rel="noopener noreferrer" target="_blank">
              <img className="mb-4 mx-4" src={imgDigitalOceanBadge} alt="Digital Ocean Forest Badge" title="DigitalOcean Forest 2021" />
            </a>
          </Col>
          {/* <Col sm md lg className="text-center">
            <Iframe
              url={'https://www.worldcommunitygrid.org/getDynamicImage.do?memberName=fabasoad&mnOn=true&stat=3&imageNum=1&rankOn=false&projectsOn=false&special=false&link=1&memberId=1103124'}
              frameBorder={0}
              name="di"
              scrolling="no"
              display="inline"
            />
          </Col> */}
        </Row>
      </Container>
    </div>
  )
}
