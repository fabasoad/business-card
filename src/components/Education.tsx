import * as React from 'react';
import { useTranslation } from 'react-i18next';

export default function Education() {
  const { t, i18n } = useTranslation();

  return (
    <div className="text-center" id="education">
      <div className="container">
        <div className="section-title center">
          <h2>{t("business-card-education-title")}</h2>
          <hr />
        </div>
        <div className="row">
          <div className="col-lg-12">
            <ul className="timeline">
              <li>
                <div className="timeline-image">
                  <h4>2010<br />-<br />2009</h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>{t("business-card-education-university-title")}</h4>
                    <h4 className="subheading">{t("business-card-education-university-master")}</h4>
                  </div>
                  <div className="timeline-body">
                    <p>{t("business-card-education-university-speciality")}</p>
                  </div>
                </div>
              </li>
              <li className="timeline-inverted">
                <div className="timeline-image">
                  <h4>2009<br />-<br />2005</h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>{t("business-card-education-university-title")}</h4>
                    <h4 className="subheading">{t("business-card-education-university-bachelor")}</h4>
                  </div>
                  <div className="timeline-body">
                    <p>{t("business-card-education-university-speciality")}</p>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}