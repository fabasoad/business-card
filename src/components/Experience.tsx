import * as React from 'react';
import { Trans, useTranslation } from 'react-i18next';

export default function Experience() {
  const { t, i18n } = useTranslation();
  return (
    <div className="text-center" id="experience">
      <div className="container">
        <div className="section-title center">
          <h2>{t("business-card-experience-title")}</h2>
          <hr />
        </div>
        <div className="row">
          <div className="col-lg-12">
            <ul className="timeline">
              <li>
                <div className="timeline-image">
                  <h4><span className="mr-1">{t("business-card-month-4")}</span><span>2020</span><br />- <br /><span className="mr-1">{t("business-card-month-7")}</span><span>2019</span></h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>Bitcoin.com</h4>
                    <h4 className="subheading">{t("business-card-experience-job-bitcoin-com-2-title")}</h4>
                  </div>
                  <div className="timeline-body">
                    <h5>{t("business-card-experience-description-title")}</h5>
                    <div>{t("business-card-experience-job-bitcoin-com-2-description")}</div>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-bitcoin-com-2-achievements-1")}</p>
                    <p>{t("business-card-experience-job-bitcoin-com-2-achievements-2")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ci")}</span><span>Travis CI, GitHub Actions, Bitrise</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-backend")}</span><span>Java 12, Kafka Streams, Spring Boot</span></p>
                  </div>
                </div>
              </li>
              <li className="timeline-inverted">
                <div className="timeline-image">
                  <h4><span className="mr-1">{t("business-card-month-6")}</span><span>2019</span><br />- <br /><span className="mr-1">{t("business-card-month-9")}</span><span>2018</span></h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>Bitcoin.com</h4>
                    <h4 className="subheading">{t("business-card-experience-job-bitcoin-com-1-title")}</h4>
                  </div>
                  <div className="timeline-body">
                    <h5>{t("business-card-experience-area-title")}</h5>
                    <p>{t("business-card-experience-job-bitcoin-com-area")}</p>
                    <h5>{t("business-card-experience-description-title")}</h5>
                    <Trans t={t}>business-card-experience-job-bitcoin-com-1-description</Trans>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-bitcoin-com-1-achievements-1")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-backend")}</span><span>Maven, Java 8 (Tomcat, Facelets, lombok, JUnit), Python</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-frontend")}</span><span>JavaScript (jQuery), Bootstrap 4</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-database")}</span><span>MariaDB, Redis</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-scm")}</span><span>Git</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ide")}</span><span>Intellij IDEA, VS Code</span></p>
                    <p> <span className="mr-1 colon-title">{t("business-card-experience-technologies-bugtracker")}</span><span>Jira</span></p>
                  </div>
                </div>
              </li>
              <li>
                <div className="timeline-image">
                  <h4><span className="mr-1">{t("business-card-month-9")}</span><span>2018</span><br />-<br /><span className="mr-1">{t("business-card-month-2")}</span><span>2016</span></h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>Lohika</h4>
                    <h4 className="subheading">{t("business-card-experience-job-lohika-3-title")}</h4>
                  </div>
                  <div className="timeline-body">
                    <h5>{t("business-card-experience-description-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-3-prysm-description-1")}</p>
                    <p>{t("business-card-experience-job-lohika-3-prysm-description-2")}</p>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-3-prysm-achievements-1")}</p>
                    <p>{t("business-card-experience-job-lohika-3-prysm-achievements-2")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-software-development")}</span><span>C#/.NET (Xamarin Forms, Prism)</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-scm")}</span><span>Git</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-bugtracker")}</span><span>Jira</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ci")}</span><span>TeamCity</span></p>
                    <p>&nbsp;</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-description-1")}</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-description-2")}</p>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-3-earnin-achievements-1")}</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-achievements-2")}</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-achievements-3")}</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-achievements-4")}</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-achievements-5")}</p>
                    <p>{t("business-card-experience-job-lohika-3-earnin-achievements-6")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-software-development")}</span><span>C#/.NET (.NET Core, Microsoft DI)</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-infrastructure-development")}</span><span>Kubernetes, helm, Docker, fluent-bit, fluentd, Elasticsearch, Kibana, Prometheus, Grafana</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-scm")}</span><span>Git</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ide")}</span><span>Visual Studio, VS Code</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-bugtracker")}</span><span>GitHub Issues</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ci")}</span><span>Jenkins</span></p>
                    <p>&nbsp;</p>
                    <p>{t("business-card-experience-job-lohika-3-general-description")}</p>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-3-general-achievements-1")}</p>
                    <p>{t("business-card-experience-job-lohika-3-general-achievements-2")}</p>
                    <p>{t("business-card-experience-job-lohika-3-general-achievements-3")}</p>
                  </div>
                </div>
              </li>
              <li className="timeline-inverted">
                <div className="timeline-image">
                  <h4><span className="mr-1">{t("business-card-month-1")}</span><span>2016</span><br />- <br /><span className="mr-1">{t("business-card-month-5")}</span><span>2013</span></h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>Lohika</h4>
                    <h4 className="subheading">{t("business-card-experience-job-lohika-2-title")}</h4>
                  </div>
                  <div className="timeline-body">
                    <h5>{t("business-card-experience-description-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-2-description")}</p>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-2-achievements-1")}</p>
                    <p>{t("business-card-experience-job-lohika-2-achievements-2")}</p>
                    <p>{t("business-card-experience-job-lohika-2-achievements-3")}</p>
                    <p>{t("business-card-experience-job-lohika-2-achievements-4")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-backend")}</span><span>Java (Spring), Maven</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-frontend")}</span><span>JavaScript (Angular)</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-database")}</span><span>MSSQL, Oracle</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-scm")}</span><span>SVN, Git</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ide")}</span><span>Intellij IDEA</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-bugtracker")}</span><span>HPE ALM</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ci")}</span><span>Jenkins</span></p>
                  </div>
                </div>
              </li>
              <li>
                <div className="timeline-image">
                  <h4><span className="mr-1">{t("business-card-month-4")}</span><span>2013</span><br />- <br /><span className="mr-1">{t("business-card-month-10")}</span><span>2011</span></h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>Lohika</h4>
                    <h4 className="subheading">{t("business-card-experience-job-lohika-1-title")}</h4>
                  </div>
                  <div className="timeline-body">
                    <h5>{t("business-card-experience-description-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-1-description")}</p>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-lohika-1-achievements-1")}</p>
                    <p>{t("business-card-experience-job-lohika-1-achievements-2")}</p>
                    <p>{t("business-card-experience-job-lohika-1-achievements-3")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-software-development")}</span><span>C#/.NET (Windows Forms)</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-scm")}</span><span>SVN</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ide")}</span><span>Visual Studio</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-bugtracker")}</span><span>HPE ALM</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ci")}</span><span>Jenkins</span></p>
                  </div>
                </div>
              </li>
              <li className="timeline-inverted">
                <div className="timeline-image">
                  <h4><span className="mr-1">{t("business-card-month-9")}</span><span>2011 </span><br />- <br /><span className="mr-1">{t("business-card-month-3")}</span><span>2010</span></h4>
                </div>
                <div className="timeline-panel">
                  <div className="timeline-heading">
                    <h4>ARKA - Finance, Software, Services</h4>
                    <h4 className="subheading">{t("business-card-experience-job-arka-title")}</h4>
                  </div>
                  <div className="timeline-body">
                    <h5>{t("business-card-experience-description-title")}</h5>
                    <p>{t("business-card-experience-job-arka-description")}</p>
                    <h5>{t("business-card-experience-achievements-title")}</h5>
                    <p>{t("business-card-experience-job-arka-achievements-1")}</p>
                    <p>{t("business-card-experience-job-arka-achievements-2")}</p>
                    <p>{t("business-card-experience-job-arka-achievements-3")}</p>
                    <h5>{t("business-card-experience-technologies-title")}</h5>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-backend")}</span><span>PHP, Codeigniter, vBulletin, OXID eShop, Smarty</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-frontend")}</span><span>C#/.NET (Silverlight), JavaScript (jQuery)</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-database")}</span><span>MySQL</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-scm")}</span><span>SVN</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-ide")}</span><span>Visual Studio 2010, NetBeans</span></p>
                    <p><span className="mr-1 colon-title">{t("business-card-experience-technologies-bugtracker")}</span><span>Redmine</span></p>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
}