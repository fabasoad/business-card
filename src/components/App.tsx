import * as React from 'react'
import { connect } from 'react-redux'
import { ThunkDispatch } from 'redux-thunk'

import AboutMain from './About/AboutMain'
import BackToTopButton from './BackToTopButton'
import CertificatesMain from './Certificates/CertificatesMain'
import EducationMain from './Education/EducationMain'
import ExperienceMain from './Experience/ExperienceMain'
import FooterMain from './Footer/FooterMain'
import Header from './Header'
import LanguagesMain from './Languages/LanguagesMain'
import MenuMain from './Menu/MenuMain'
import PortfolioMain from './Portfolio/PortfolioMain'
import Resume from './Resume'
import Skills from './Skills'
import { Locale, AppActions } from '../store/locale/types'
import { AppState } from '../store/configureStore'

interface AppProps {
}

type Props = AppProps & LinkDispatchProps & LinkStateProps

export function App(props: Props) {
  return (
    <div className={`font-regular font-${props.locale.code === 'jp' ? '' : 'non-'}jp`}>
      <BackToTopButton />
      <Header />
      <MenuMain />
      <AboutMain />
      <Resume />
      <Skills />
      <ExperienceMain />
      <EducationMain />
      <CertificatesMain />
      <LanguagesMain />
      <PortfolioMain />
      <FooterMain />
    </div>
  )
}

interface LinkStateProps {
  locale: Locale
}

interface LinkDispatchProps {
}

const mapStateToProps = (
  state: AppState,
  ownProps: AppProps
): LinkStateProps => ({
  locale: state.locale
})

const mapDispatchToProps = (
  dispatch: ThunkDispatch<any, any, AppActions>,
  ownProps: AppProps
): LinkDispatchProps => ({
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
