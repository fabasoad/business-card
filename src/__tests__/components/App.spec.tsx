/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { App } from '../../components/App'

new Array<string>('jp', 'gb').forEach((l) =>
  test(`should render App with ${l} locale correctly`, () => {
    const wrapper: ShallowWrapper = shallow(
      <App locale={{ code: l, title: '' }} />
    )
    expect(wrapper).toMatchSnapshot()
  })
)
