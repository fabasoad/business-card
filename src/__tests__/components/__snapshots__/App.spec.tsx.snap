// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App with gb locale correctly 1`] = `
<div
  className="font-regular font-non-jp"
>
  <BackToTopButton />
  <Header />
  <MenuMain />
  <Connect(AboutMain) />
  <StatsMain
    autoload={true}
  />
  <Resume />
  <Skills />
  <ExperienceMain />
  <EducationMain />
  <CertificatesMain />
  <LanguagesMain />
  <PortfolioMain />
  <Badges />
  <FooterMain />
</div>
`;

exports[`should render App with jp locale correctly 1`] = `
<div
  className="font-regular font-jp"
>
  <BackToTopButton />
  <Header />
  <MenuMain />
  <Connect(AboutMain) />
  <StatsMain
    autoload={true}
  />
  <Resume />
  <Skills />
  <ExperienceMain />
  <EducationMain />
  <CertificatesMain />
  <LanguagesMain />
  <PortfolioMain />
  <Badges />
  <FooterMain />
</div>
`;
